### PR TITLE
Add validation pattern

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: BackofficeBundle/Validator/Constraints/BlockNodePatternValidator.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                        $this->translator->trans('open_orchestra_backoffice.form.node.error.pattern', array(
+                            '%blockLabel%' => $blockLabel,
+                            '%parameter%' => $parameter))
+                    );
+                    $isValid = false;
+                }
+            }
+        }
+        if (!$isValid) {
+            $response = $this->templating->renderResponse('BraincraftedBootstrapBundle::flash.html.twig');
```
Comment (nicolasThal) [inline BackofficeBundle/Validator/Constraints/BlockNodePatternValidator.php]: you should not use the renderResponse, the render method will render you only the response content

**Comment by nicolasThal** [comment]
Did you forget to commit the tests?

**Comment by itkg-nanne** [comment]
@nicolasThal , I'm sure, you know the answer ;)
As we spoke about the way to do this ticket, i send this PR as propositionnal PR.
I'm actually doing the tests during your returns.

**Comment by nicolasThal** [comment]
the way is quite ok for me

**Comment by itkg-nanne** [comment]
test done @nicolasThal

File: Backoffice/GenerateForm/Strategies/ContentStrategy.php [hunk 2]
```diff
@@ -40,6 +40,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         ));
     }
 
+    public function getRequiredUriParameter()
```
Comment (nicolasThal) [inline Backoffice/GenerateForm/Strategies/ContentStrategy.php]: method comments

File: BackofficeBundle/StrategyManager/GenerateFormManager.php [hunk 3]
```diff
@@ -57,6 +57,25 @@ public function getDefaultConfiguration(BlockInterface $block)
...
+    public function getRequiredUriParameter(BlockInterface $block)
+    {
+        /** @var GenerateFormInterface $strategy */
+        foreach ($this->strategies as $strategy) {
+            if ($strategy->support($block)) {
+                return $strategy->getRequiredUriParameter();
+            }
+        }
+
+        return array();
```
Comment (nicolasThal) [inline BackofficeBundle/StrategyManager/GenerateFormManager.php]: as you implement `return array()` in the AbstractBlockStrategy for the getRequiredUriParameter, there is no need to return an array here, you should throw an exception I think

Comment (itkg-nanne) [inline BackofficeBundle/StrategyManager/GenerateFormManager.php]: Why not, but this way is used in getDEfaultParameter. So we should change it too or keep it like that, @nicolasThal

Comment (nicolasThal) [inline BackofficeBundle/StrategyManager/GenerateFormManager.php]: I think we should change it too

**Comment by itkg-nanne** [comment]
done @nicolasThal

File: BackofficeBundle/StrategyManager/GenerateFormManager.php [hunk 4]
```diff
@@ -53,7 +55,26 @@ public function getDefaultConfiguration(BlockInterface $block)
             }
         }
 
-        return array();
+        throw new MissingGenerateFormStrategyException();
```
Comment (nicolasThal) [inline BackofficeBundle/StrategyManager/GenerateFormManager.php]: you should also change the method comments

File: BackofficeBundle/StrategyManager/GenerateFormManager.php [hunk 5]
```diff
@@ -53,7 +55,26 @@ public function getDefaultConfiguration(BlockInterface $block)
...
+    }
+
+    /**
+     * Get the required Uri parameters for the block
+     *
+     * @param BlockInterface $block
+     *
+     * @return array
```
Comment (nicolasThal) [inline BackofficeBundle/StrategyManager/GenerateFormManager.php]: you should add the @throws annotation

**Comment by nicolasThal** [comment]
could you also update the `GenerateFormManagerTest` ?

**Comment by itkg-nanne** [comment]
done @nicolasThal

File: BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php [hunk 6]
```diff
@@ -69,4 +62,24 @@ public function testCreateForm()
         $strategy = $this->manager->createForm($this->block);
         $this->assertSame($this->strategy1, $strategy);
     }
+
+    /**
+     * Test getDefaultConfiguration
+     */
+    public function testGetDefaultConfiguration() {
```
Comment (nicolasThal) [inline BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php]: psr

File: BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php [hunk 7]
```diff
@@ -69,4 +62,24 @@ public function testCreateForm()
...
+        $this->manager->getDefaultConfiguration($this->block);
+
+        Phake::verify($this->strategy1)->getDefaultConfiguration();
+        Phake::verify($this->strategy2, Phake::never())->getDefaultConfiguration();
+    }
+
+    /**
+     * Test getDefaultConfiguration
+     */
+    public function testGetRequiredUriParameter() {
```
Comment (nicolasThal) [inline BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php]: psr

File: BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php (hunk 6 above)
Comment (nicolasThal) [inline BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php]: the tests which checks that the exceptions has been thrown are not done ?

Comment (itkg-nanne) [inline BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php]: never done, to be done.... ;) @nicolasThal

Comment (nicolasThal) [inline BackofficeBundle/Tests/StrategyManager/GenerateFormManagerTest.php]: :), could you write a ticket for it then ?

**Comment by itkg-nanne** [comment]
done @nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**